cmd/thv-operator/pkg/git/client.go

```diff
@@ -0,0 +1,203 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+// Client defines the interface for Git operations
+type Client interface {
+	// Clone clones a repository with the given configuration
+	Clone(ctx context.Context, config *CloneConfig) (*RepositoryInfo, error)
+
+	// Pull updates an existing repository
+	Pull(ctx context.Context, repoInfo *RepositoryInfo) error
+
+	// GetFileContent retrieves the content of a file from the repository
+	GetFileContent(repoInfo *RepositoryInfo, path string) ([]byte, error)
+
+	// GetCommitHash returns the current commit hash
+	GetCommitHash(repoInfo *RepositoryInfo) (string, error)
+
+	// Cleanup removes local repository directory
+	Cleanup(repoInfo *RepositoryInfo) error
+}
+
+// DefaultGitClient implements GitClient using go-git
+type DefaultGitClient struct{}
+
+// NewDefaultGitClient creates a new DefaultGitClient
+func NewDefaultGitClient() *DefaultGitClient {
+	return &DefaultGitClient{}
+}
+
+// Clone clones a repository with the given configuration
+func (c *DefaultGitClient) Clone(ctx context.Context, config *CloneConfig) (*RepositoryInfo, error) {
+	// Prepare clone options (no authentication for initial version)
+	cloneOptions := &git.CloneOptions{
```

Comment:
Doesn't this fetch the whole history?

I'm fine acking this if we file follow-up tickets, but what I would prefer is to make a shallow clone if possible with just the commit we are interested in. Using `Depth: 1` could go a long way. Another thing that would be nice is to have a hard limit on the data we fetch. This could be done either with a separate filesystem (we used billyfs in a past project, see https://github.com/mindersec/minder/blob/800e9186023bb92d3533cf2ada5a73e925c587e2/internal/providers/git/git.go#L33).